packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm

```diff
@@ -469,8 +469,10 @@ - (void)_loadScriptFromSource:(RCTSource *)source
   [[NSNotificationCenter defaultCenter] postNotificationName:@"RCTInstanceDidLoadBundle" object:nil];
 
   if (_onInitialBundleLoad) {
-    _onInitialBundleLoad();
-    _onInitialBundleLoad = nil;
+    auto onInitialBundleLoad = _onInitialBundleLoad;
+    RCTExecuteOnMainQueue(^{
+      onInitialBundleLoad();
+    });
```

Comment:
We are not nullifying the bundle load anymore...
can we instead (or something similar, syntax could be wrong):
```suggestion
		__weak typeof(self) weakSelf = self; 
    RCTExecuteOnMainQueue(^{
    	__strong typeof(self) strongSelf = weakSelf; 
			strongSelf._onInitialBundleLoad();
			strongSelf._onInitialBundleLoad = nil
    });
```

Reply:
@cipolleschi  Oops, my bad, I missed it. Already re-added. :)